.github/workflows/publish.yml

```diff
@@ -1,87 +0,0 @@
-# Publishes built jars to distribution platforms
-name: Publish
-
-on:
-  push:
-    tags:
-      - 'v[0-9]+.[0-9]+.[0-9]+' # any semver tag, e.g. v1.2.3
-
-env:
-  # link to the changelog with a format code for the version
-  CHANGELOG_LOCATION: "Changelog is available [here](https://github.com/${{github.repository}}/releases/tag/${{github.ref_name}})"
-  # type of release
-  RELEASE_TYPE: "beta"
-
-jobs:
-  Publish:
-    runs-on: ubuntu-latest
-
-    permissions: 
-      contents: write # needed to create GitHub releases
-
-    steps:
-      - name: Checkout Repository
-        uses: actions/checkout@v3
-      
-      - name: Check for Duplicate Tags
-        run: |
-          if git rev-parse -q --verify "refs/tags/${{ github.ref }}" >/dev/null; then
-            echo "Tag already exists. A version bump is required."
-            exit 1
-          fi
```

Comment:
Should we keep this check for missing a version bump? It seems like a nice safeguard to have

Reply:
Since we no longer manually specify the version tag to publish, this is no longer needed. Previously, we had GitHub create a release which as a result creates a git tag with the specified version, which could overwrite releases due to how the github API worked (I assume).

Now, we push the version tag ourselves. Git will automatically reject attempts to push duplicate tags, so we don't need to check it ourselves in the workflow anymore. It also means we don't have to fetch the entire git history when cloning, which should speed things up a bit.